[PATCH 1/7] plat: mediatek: define macros for identifying boot device

Define macros for identifying boot device in C source code

Change-Id: I2a190e765b936db493c5ec4ccd81fbd3778d8d74
---
 plat/mediatek/apsoc_common/bl2/bl2_image.mk | 10 ++++++++++
 1 file changed, 10 insertions(+)
diff --git a/plat/mediatek/apsoc_common/bl2/bl2_image.mk b/plat/mediatek/apsoc_common/bl2/bl2_image.mk
index 78b721a2b..b58ac648f 100644

Change-Id: I2a190e765b936db493c5ec4ccd81fbd3778d8d74
--- a/plat/mediatek/apsoc_common/bl2/bl2_image.mk
+++ b/plat/mediatek/apsoc_common/bl2/bl2_image.mk
@@ -32,6 +32,7 @@ $(call DUAL_FIP_UNSUPPORTED)
 BL2_SOURCES		+=	drivers/io/io_memmap.c				\
 				$(APSOC_COMMON)/bl2/bl2_boot_ram.c
 BROM_HEADER_TYPE	:=	nor
+BL2_CPPFLAGS		+=	-DBL2_BOOT_RAM
 ifeq ($(RAM_BOOT_DEBUGGER_HOOK), 1)
 BL2_CPPFLAGS		+=	-DRAM_BOOT_DEBUGGER_HOOK
 endif
@@ -50,6 +51,7 @@ $(call BL2_FIP_OVERRIDE_COMMON)
 BL2_SOURCES		+=	drivers/io/io_memmap.c				\
 				$(APSOC_COMMON)/bl2/bl2_boot_nor_mmap.c
 BROM_HEADER_TYPE	:=	nor
+BL2_CPPFLAGS		+=	-DBL2_BOOT_NOR_MMAP
 endef # End of BL2_BOOT_NOR_MMAP

 define BL2_BOOT_NOR
@@ -63,6 +65,7 @@ BL2_SOURCES		+=	drivers/mtd/nor/spi_nor.c			\
 BL2_CPPFLAGS		+=	-Ilib/libfdt
 BL2_CPPFLAGS		+=	-DMTK_SPIM_NOR
 BROM_HEADER_TYPE	:=	nor
+BL2_CPPFLAGS		+=	-DBL2_BOOT_NOR
 endef # End of BL2_BOOT_NOR

 define BL2_BOOT_MMC
@@ -76,16 +79,19 @@ BL2_SOURCES		+=	drivers/mmc/mmc.c				\
 BL2_CPPFLAGS		+=	-I$(APSOC_COMMON)/drivers/mmc			\
 				-DMTK_MMC_BOOT
 BL2_CFLAGS		+=	-march=armv8-a+crc
+BL2_CPPFLAGS		+=	-DBL2_BOOT_MMC
 endef # End of BL2_BOOT_MMC

 define BL2_BOOT_EMMC
 $(call BL2_BOOT_MMC)
 BROM_HEADER_TYPE	:=	emmc
+BL2_CPPFLAGS		+=	-DBL2_BOOT_EMMC
 endef # End of BL2_BOOT_EMMC

 define BL2_BOOT_SD
 $(call BL2_BOOT_MMC)
 BROM_HEADER_TYPE	:=	sdmmc
+BL2_CPPFLAGS		+=	-DBL2_BOOT_SD
 endef # End of BL2_BOOT_SD

 #
@@ -125,6 +131,7 @@ endif
 ifneq ($(NMBM_DEFAULT_LOG_LEVEL),)
 BL2_CPPFLAGS		+=	-DNMBM_DEFAULT_LOG_LEVEL=$(NMBM_DEFAULT_LOG_LEVEL)
 endif
+BL2_CPPFLAGS		+=	-DBL2_BOOT_SPI_NAND_NMBM
 endif

 ifeq ($$(UBI),1)
@@ -137,6 +144,7 @@ BL2_CFLAGS		+=	-march=armv8-a+crc
 else
 BL2_SOURCES		+=	drivers/io/ubi/crc32.c
 endif
+BL2_CPPFLAGS		+=	-DBL2_BOOT_SPI_NAND_UBI
 endif

 ifneq ($$(NMBM),1)
@@ -172,6 +180,7 @@ BL2_SOURCES		+=	$$(MTK_SNAND_SOURCES)				\
 BL2_CPPFLAGS		+=	-I$(APSOC_COMMON)/drivers/snfi
 BL2_CPPFLAGS		+=	-DPRIVATE_MTK_SNAND_HEADER
 BROM_HEADER_TYPE	:=	snand
+BL2_CPPFLAGS		+=	-DBL2_BOOT_SNFI
 endef # End of BL2_BOOT_SNFI

 #
@@ -186,6 +195,7 @@ BL2_SOURCES		+=	drivers/mtd/nand/core.c				\
 BL2_CPPFLAGS		+=	-Ilib/libfdt
 BL2_CPPFLAGS		+=	-I$(APSOC_COMMON)/drivers/spi_nand
 BROM_HEADER_TYPE	:=	spim-nand
+BL2_CPPFLAGS		+=	-DBL2_BOOT_SPI_NAND
 endef # End of BL2_BOOT_SPI_NAND

 #
--
2.45.2

